it/pages/agent/ConfirmEmailPageSpec.scala

```diff
@@ -0,0 +1,182 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pages.agent
+
+import helpers.IntegrationTestConstants._
+import pages.BasePageISpec
+import play.api.libs.ws.WSResponse
+import play.api.test.Helpers._
+import stubs.EmailVerificationStub
+
+class ConfirmEmailPageSpec extends BasePageISpec {
+
+  val path = "/confirm-agent-email"
+
+  "Calling the .show action" when {
+
+    "the user is an Agent" when {
+
+      "the Agent is signed up for HMRC-AS-AGENT" when {
+
+        "there is no notification email in session" should {
+
+          "Render the Confirm Email view" in {
+
+            def show(): WSResponse = get(path, formatNotificationsEmail(None) ++ formatReturnUrl)
+
+            given.agent.isSignedUpToAgentServices
+
+            When("I call the Confirm email page with no notification email in session")
+
+            val res = show()
+
+            res should have(
+              httpStatus(SEE_OTHER),
+              redirectURI(controllers.agent.routes.CapturePreferenceController.show().url)
+            )
+          }
+        }
+
+        "there is a notification email in session" should {
+
+          "Render the Confirm Email view" in {
+
+            def show(): WSResponse = get(path, formatNotificationsEmail(Some(notificationsEmail)) ++ formatReturnUrl)
+
+            given.agent.isSignedUpToAgentServices
+
+            When("I call the Confirm email page with a notification email in session")
+            val res = show()
+
+            res should have(
+              httpStatus(OK),
+              elementText("h1")("Confirm the new email address")
+            )
+          }
+        }
+      }
+    }
+  }
+
+  "Calling the .isEmailVerified action" when {
+
+    "the user is an Agent" when {
+
+      "the Agent is signed up for HMRC-AS-AGENT" when {
+
+        "there is a verified notification email in session" should {
+
+          "Render the Select client VRN view" in {
+
+            val path = "/is-email-verified"
+
+            def isEmailVerified(): WSResponse = get(path, formatNotificationsEmail(Some(notificationsEmail)) ++
+              formatReturnUrl)
+
+            given.agent.isSignedUpToAgentServices
+
+            And("I stub an email verified successful response from EmailVerificationService")
+            EmailVerificationStub.stubEmailVerified
+
+            When("I call the isEmailVerified controller action with a verified notification email in session")
+
+            val res = isEmailVerified()
+
+            res should have(
+              httpStatus(SEE_OTHER),
+              redirectURI(controllers.agent.routes.SelectClientVrnController.show().url)
```

Comment:
What do you think about having a title / header check here as well? Just to make sure that the correct view is loaded?

Reply:
Is that possible on a redirect? I did try but couldn't find any html body.

Reply:
Good point actually. Ignore it then